### DISCUSSION TITLE
Datagird In Tabs Won't Load data on Init

### DISCUSSION THREAD

**Discussion by peterhym21** [Megabit/Blazorise#3864] [Q&A]
Hi when ever the page fist loads all the data in the datagrid wont load in before I click on the tabs that is the selected tab, after that it loads pefectly fin even when moving between pages, but that fist time i just wont load 
How can I get the Data to load on init

        <Tabs RenderMode="TabsRenderMode.LazyReload" SelectedTab="tab1" Justified="true">
            <Items>
                <Tab Name="tab1">Operations</Tab>
                <Tab Name="tab2">Assembly Function</Tab>
                <Tab Name="tab3">Workplace Function</Tab>
                <Tab Name="tab4">Operation Types</Tab>
            </Items>
            <Content>
                <TabPanel Name="tab1">
                    @if (Operations is not null)
                    {
                        <DataGrid TItem="UiOperationListVm"
                              Data="@Operations"
                              @bind-SelectedRow="@SelectedOperation"
                              EditMode="DataGridEditMode.Inline"
                              Editable
                              PageSize="10"
                              ShowPager
                              Responsive
                              UseValidation>

                            <DataGridColumn Field="@nameof(UiOperationListVm.Id)" Caption="ID" Sortable="false" Displayable="true" />
                            <DataGridColumn Field="@nameof(UiOperationListVm.Name)" Caption="Name" Editable />
                            <DataGridColumn Field="@nameof(UiOperationListVm.Description)" Caption="Description" Editable />
                            <DataGridColumn Field="OperationType.Description" Caption="Operation types" Editable>
                                <EditTemplate>
                                    <SelectList TItem="UiOperationTypeListVm"
                                            TValue="int"
                                            Data="@OperationTypeListVm"
                                            TextField="@((item)=>item.Description)"
                                            ValueField="@((item)=>(int)item.Name)"
                                            SelectedValue="@UpdateOperationCommand.OperationTypeName"
                                            SelectedValueChanged="@MyListValueChangedHandlerOperationsUpdate"
                                            DefaultItemText="select a Operation Type"
                                            class="btn btn-secondary dropdown-toggle" />

                                </EditTemplate>
                            </DataGridColumn>
                            <DataGridCommandColumn>
                                <EditCommandTemplate>
                                    <Button style="margin:10px;" Color="Color.Primary" Clicked="@context.Clicked">Edit</Button>
                                </EditCommandTemplate>

                                <DeleteCommandTemplate>
                                    <Button style="margin:10px;" Color="Color.Danger" Clicked="@(()=>ShowDeleteOperationMessage(context))">@context.LocalizationString</Button>
                                </DeleteCommandTemplate>

                                <SaveCommandTemplate>
                                    <Button style="margin:10px;" Color="Color.Primary" Clicked="@(()=>OnSaveClickedOperation(context))">@context.LocalizationString</Button>
                                </SaveCommandTemplate>

                                <CancelCommandTemplate>
                                    <Button style="margin:10px;" Color="Color.Danger" Clicked="@context.Clicked">Cancel</Button>
                                </CancelCommandTemplate>

                                <NewCommandTemplate></NewCommandTemplate>

                            </DataGridCommandColumn>

                        </DataGrid>
                    }
                </TabPanel>
                <TabPanel Name="tab2">
                    <TextEdit></TextEdit>
                </TabPanel>
                <TabPanel Name="tab3">
                    <TextEdit></TextEdit>
                </TabPanel>
                <TabPanel Name="tab4">
                    <TextEdit></TextEdit>
                </TabPanel>
            </Content>
        </Tabs>

**Comment by David-Moreira** [comment]
Are you sure it's not because of `Operations is not null` at the time it's rendered?
I can't recall now, but I think `DataGrid` handles null as an empty collection. You can try that by just removing the  `@if (Operations is not null)`

Or make `Operations` be an empty collection when it's null.

**Reply by peterhym21** [reply]
Well if I remove the `@if (Operations is not null)` statment, it only loads the tabel but the data in the tabel. it is still not loading in
But maby a DB problem

But it does fix som of it so thanks 👍🏼

**Reply by David-Moreira** [reply]
Alright, if you're still having problems solving this, if you can somehow recreate this example, but with the supposed data loading of the `Operations`, we can try to help. :)

**Reply by peterhym21** [reply]
I think I have found the problem

When looking at the console logs the Get DataGrid is run before the data from the DB is rettrevied:

```
2022-06-03 11:04:03.523 [INF] Entity Framework Core "6.0.5" initialized '"MccDbContext"' using provider '"Microsoft.EntityFrameworkCore.SqlServer":"6.0.3"' with options: "None"
2022-06-03 11:04:04.444 [INF] HTTP "GET" "/_content/Blazorise/textEdit.js" responded 304 in 1.4176 ms
2022-06-03 11:04:04.461 [INF] HTTP "GET" "/_content/Blazorise/validators/NumericMaskValidator.js" responded 304 in 0.2788 ms
2022-06-03 11:04:04.464 [INF] HTTP "GET" "/_content/Blazorise/validators/RegExMaskValidator.js" responded 304 in 0.4296 ms
2022-06-03 11:04:04.466 [INF] HTTP "GET" "/_content/Blazorise/validators/NoValidator.js" responded 304 in 0.3192 ms
2022-06-03 11:04:04.464 [INF] HTTP "GET" "/_content/Blazorise/validators/DateTimeMaskValidator.js" responded 304 in 0.4176 ms
**2022-06-03 11:04:04.468 [INF] HTTP "GET" "/_content/Blazorise.DataGrid/datagrid.js" responded 304 in 0.2892 ms**
2022-06-03 11:04:04.536 [INF] Executed DbCommand ("119"ms) [Parameters=[""], CommandType='Text', CommandTimeout='30']"\r\n""SELECT [o].[Name], [o].[Description], [o].[Group]\r\nFROM [OperationTypes] AS [o]"
2022-06-03 11:04:04.841 [INF] Executed DbCommand ("8"ms) [Parameters=[""], CommandType='Text', CommandTimeout='30']"\r\n""SELECT [o].[Id], [o].[CreatedBy], [o].[CreatedDate], [o].[Description], [o].[LastModifiedBy], [o].[LastModifiedDate], [o].[Name], [o].[OperationSpecialId], [o].[OperationTypeName], [o0].[Name], [o0].[Description], [o0].[Group]\r\nFROM [Operations] AS [o]\r\nINNER JOIN [OperationTypes] AS [o0] ON [o].[OperationTypeName] = [o0].[Name]"
2022-06-03 11:04:04.934 [INF] Executed DbCommand ("3"ms) [Parameters=[""], CommandType='Text', CommandTimeout='30']"\r\n""SELECT [a].[Id], [a].[CreatedBy], [a].[CreatedDate], [a].[Description], [a].[LastModifiedBy], [a].[LastModifiedDate], [a].[MustCompleteAll], [a].[MustCompleteLast], [a].[MustStartOnFirst], [a].[Name]\r\nFROM [AssemblyFunctions] AS [a]"
```
